src/lib_json/json_reader.cpp

```diff
@@ -103,7 +103,7 @@ Reader::Reader(const Features& features)
 
 bool
 Reader::parse(const std::string& document, Value& root, bool collectComments) {
-  JSONCPP_STRING documentCopy(document.data(), document.data() + document.capacity());
+  JSONCPP_STRING documentCopy(document.data(), document.data() + document.length());
```

Comment:
`JSONCPP_STRING` is going to be some kind of `std::basic_string`. The allocator changes with different build configurations. So it has the full `std::string` constructor API.

    JSONCPP_STRING documentCopy(document.begin(), document.end());

But really we don't need `documentCopy` at all. The swap doesn't seem to be doing anything we can't do better with an assignment.

    document_.assign(document.begin(), document.end());